tools/dragen-somatic/4.3.6/dragen-somatic__4.3.6.cwl

```diff
@@ -1463,6 +1463,48 @@ inputs:
       prefix: "--repeat-genotype-use-catalog="
       separate: False
 
+  # Cross-sample Contamination
+  qc_somatic_contam_vcf:
+    label: qc somatic contam vcf
+    doc: |
+      The cross-contamination metric is enabled by including one of the following flags along with a compatible VCF.
+      Pre-built contamination VCF files for different human references can be found at /opt/edico/config. 
+      DRAGEN supports separate modes for germline and somatic samples. 
+      For hg38 references, use somatic_sample_cross_contamination_resource_hg38.vcf.gz
+      For hg19 references, use somatic_sample_cross_contamination_resource_hg19.vcf.gz
+      For GRCh37 references, use somatic_sample_cross_contamination_resource_GRCh37.vcf.gz
+    type:
+      - "null"
+      - File
+      - type: enum
+        symbols:
+          - "somatic_sample_cross_contamination_resource_hg38.vcf.gz"
+          - "somatic_sample_cross_contamination_resource_hg19.vcf.gz"
+          - "somatic_sample_cross_contamination_resource_GRCh37.vcf.gz"
+    inputBinding:
+      prefix: "--qc-somatic-contam-vcf="
+      separate: False
+      valueFrom: >-
+        ${
```

Comment:
Great work here! 

One thing you could do is set the enums list to be just
* hg38, hg19, GRCh37 and then use the js to extend the string to `somatic_sample_cross_contamination_resource_hg38.vcf.gz`, makes it easier for the user to select. 

In dragen tools, then add in the functions get_somatic_cross_contamination_resource() which accepts an input of type 'file' or of type string and resolves the file to self.path or the enum to get_contamination_dir() + "somatic_cross_contamination_resource_" + self etc

Might save a bit of wet typing between dragen versions, and would mean an abstracted function could work for both germline and somatic paths

Reply:
updated in [`c8b2b35` (#618)](https://github.com/umccr/cwl-ica/pull/618/commits/c8b2b35577364c1e4422ba4506a60a661fe08721)